### PR TITLE
Pass raw ID data

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Passing along the raw data for IDs](https://trello.com/c/68Vbflqn/513-passing-along-the-raw-data-for-ids)

File: lib/customers.js [hunk 1]
```diff
@@ -281,6 +281,18 @@ function enhanceAtFields (fields) {
...
+    'id_card_data'
+  ]
+
+  const fieldsToUpdate = _.pick(fieldsWithRawData, fields)
+  const rawFields = _.fromPairs(_.map(f => [`${f}_raw`, JSON.stringify(fieldsToUpdate[f].raw)], _.keys(fieldsToUpdate)))
+  const fieldsWithoutRawData = _.mapValues(f => _.omit(['raw'], f), fieldsToUpdate)
+
+  return _.merge(fieldsWithoutRawData, rawFields)
+}
+
```
Comment (RafaelTaranto) [inline lib/customers.js]: isn't this removing any other fields and just updating id_card_data?

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: lib/customers.js (hunk 1 above)
Comment (chaotixkilla) [reply lib/customers.js]: You are correct! This next commit should solve this, nice catch

File: lib/customers.js [hunk 2]
```diff
@@ -281,6 +281,20 @@ function enhanceAtFields (fields) {
...
+function updateRawData (fields) {
+  const fieldsWithRawData = [
+    'id_card_data'
+  ]
+
+  const fieldsToUpdate = _.intersection(fieldsWithRawData, _.keys(fields))
+  const rawFields = _.fromPairs(_.map(f => [`${f}_raw`, JSON.stringify(fields[f].raw)], fieldsToUpdate))
+  const fieldsWithoutRawData = _.mapValues(f => {
+    return _.has(f, 'raw') ? _.omit(['raw'], f) : f
+  }, fields)
```
Comment (RafaelTaranto) [inline lib/customers.js]: this still has the potential of breaking fields not on fieldsWithRawData. What about something like:

```js
const fieldsWithRawData = [
  'id_card_data'
]

const mapPairToRaw = _.map(([key, value]) => ([
  [key, _.omit(['raw'], value)],
  [`${key}_raw`, JSON.stringify(value.raw)]
]))

function updateRawData (fields) {
  const pickTransformable = _.pick(fieldsWithRawData)
  const extractRaws = _.compose(_.fromPairs, _.flatten, mapPairToRaw, _.toPairs)
  const assignOriginal = _.assign(fields)

  return _compose(assignOriginal, extractRaws, pickTransformable)(fields)
}
```

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**